drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -130,59 +148,115 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
 static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
 {
     at86rf2xx_t *dev = (at86rf2xx_t *)netdev;
-    uint8_t phr;
     size_t pkt_len;
 
+#ifndef MODULE_AT86RFR2
+    uint8_t phr;
+#endif
+
     /* frame buffer protection will be unlocked as soon as at86rf2xx_fb_stop()
      * is called*/
     at86rf2xx_fb_start(dev);
 
     /* get the size of the received packet */
+#ifdef MODULE_AT86RFR2
+    /* Transceiver Received Frame Length Register (refer p.35, 85, 154) */
+    /* subtract length of FCS */
+    pkt_len = (TST_RX_LENGTH & 0x7f) - 2;
+#else
     at86rf2xx_fb_read(dev, &phr, 1);
 
-    /* ignore MSB (refer p.80) and substract length of FCS field */
+    /* ignore MSB (refer p.80) and subtract length of FCS field */
     pkt_len = (phr & 0x7f) - 2;
+#endif
 
-    /* just return length when buf == NULL */
+    /* just return length when buf == NULL, Upper layer calls _recv
+     * with no buffer to get expected size for allocating buffer
+     *
+     * Attetion:fb_stop does remove the frame buffer protection !
+     * */
     if (buf == NULL) {
+        /* TODO Is this not a problem? Frame Buffer release */
         at86rf2xx_fb_stop(dev);
+        DEBUG("Packet lenght: %d\n", pkt_len);
```

Comment:
@miri64 or @thomaseichinger, do you have an opinion on this?

Reply:
To clarify the progblem. this function gets called 2 times, one time to get back the size for the buffer and one time to fetch the data. But the frame bufer protection is allready released after the size is determined. Which could result in data loss when a node send data meanwhile.

Reply:
The expected behavior is well defined in http://api.riot-os.org/structnetdev__driver.html#ae2c8cad80067e3b1f9979931ddb3cc8b
I agree there is a problem with the current implementation, this should be fixed for all variants, but in a separate PR. I think the proper fix is to only call at86rf2xx_fb_stop() when len > 0 (see docs for netdev_driver_t recv

Reply:
I will prepare this, can we merge that beforehand, so that a can adjust the at86rfr2 driver rebased on that?